ls1intum/Artemis
# Enhancement/programming exercise/build failed property

### Checklist

### Motivation and Context


Resolves #1153 

We currently rely on a hardcoded String comparison every time we determine whether a result for a submission has a build error (for programmin exercises)

### Description

As discussed with @krusche : 
- I added a new boolean `buildFailed` to the `ProgrammingSubmission` class.
- I migrated the `buildArtifact` boolean to the `ProgrammingSubmission` class, so the `Result` class does not contain this programming exercise specific attribute.

I also added all necessary migration scripts.

### Steps for Testing


1. Log in to Artemis
2. Participate in a programming exercise and produce a build error
3. This status should be reflected in all result components (code-editor, overview, scores, etc.)
